bigquery_etl/cli/check.py

```diff
@@ -0,0 +1,199 @@
+"""bigquery-etl CLI check command."""
+import datetime
+import re
+import subprocess
+import sys
+import tempfile
+from subprocess import CalledProcessError
+
+import click
+
+from ..cli.utils import is_authenticated, paths_matching_checks_pattern, sql_dir_option
+from ..util.common import render as render_template
+
+
+def _build_query_arguments(parameters):
+    """
+    Convert a dict of parameters (from CLI) into query arguments.
+
+    Opposite of _build_query_arguments.
+    """
+    if parameters is None:
+        return []
+    return [f"--parameter={key}::{value}" for key, value in parameters.items()]
+
+
+def _parse_query_arguments(query_args):
+    parameters = []
+    for query_arg in query_args:
+        if query_arg.startswith("--parameter=") and len(query_arg.split("=")) == 2:
+            parameters.append(query_arg.split("=")[1])
+        else:
+            raise ValueError("argument must start with --parameter")
+    return _build_parameters(None, None, parameters)
+
+
+def _build_parameters(ctx=None, param=None, parameters=None):
+    """
+    Convert the list of parameters from format <param-name>::<param-value> to dict.
+
+    If an invalid parameter is found, empty dict is returned.  Opposite of _build_query_arguments.
+    """
+    result = {}
+    detected_partition = False
+
+    if parameters is None:
+        return {}
+
+    for param in parameters:
+        params = param.split(":")
+        if len(params) != 3:
+            raise ValueError(
+                f"parameter: {param} is not a valid parameter.  Please use format: <param-name>::<param-value>, exiting"
+            )
```

Comment:
By splitting the parameter definition we lose the type information (`params[1]`)

Reply:
No longer doing this.